src/linter.cc

```diff
@@ -24,93 +24,235 @@
+            // ignore the line
+            while ( i < static_cast<int>(sql.size()) &&
+                    sql[i] != '\n' ) {
+                ++i;
+            }
+        }
 
-    // KeywordExtractor(output->statement(), keywords).extract();
+        if (!inside_string && sql[i-1] == '/' && sql[i] == '/') {
+            includes_type2 = true;
+            // ignore the line
+            while ( i < static_cast<int>(sql.size()) &&
+                    sql[i] != '\n' ) {
+                ++i;
+            }
+        }
 
-    for (const zetasql::ASTNode *keyword : *keywords) {
-        if ( !allUpperCase(keyword) ) {
-            return absl::Status(
-                absl::StatusCode::kFailedPrecondition, "");
+        // ignore multiline comments
+        if (!inside_string && sql[i-1] == '/' && sql[i] == '*') {
+            // it will start checking after '/*' and after the iteration
+            // finished, the pointer 'i' will be just after '*/' (incrementation
+            // from the for statement is included)
+            i += 2;
+            while ( i < static_cast<int>(sql.size()) &&
+                    !(sql[i-1] == '*' && sql[i] == '/') ) {
+                ++i;
+            }
         }
+
+        if (sql[i] == '\'' || sql[i] == '"')
+            inside_string = !inside_string;
     }
 
+    if ( includes_type1 && includes_type2 )
+        return absl::Status(
+                absl::StatusCode::kFailedPrecondition,
+                absl::StrCat("either '//' or '--' should be used to ",
+                            "specify a comment"));
     return absl::OkStatus();
 }
 
+absl::Status ASTNodeRule::ApplyTo(absl::string_view sql) {
+    RuleVisitor visitor(rule_, sql);
+    absl::Status return_status = absl::OkStatus();
+    std::unique_ptr<zetasql::ParserOutput> output;
+
+    zetasql::ParseResumeLocation location =
+        zetasql::ParseResumeLocation::FromStringView(sql);
+    bool is_the_end = false;
+    while ( !is_the_end && return_status.ok() ) {
+        return_status = zetasql::ParseNextScriptStatement(
+            &location, zetasql::ParserOptions(), &output, &is_the_end);
+        if ( return_status.ok() ) {
+            return_status = output->statement()->TraverseNonRecursive(&visitor);
+        }
+    }
+    if ( return_status.ok() ) {
+        return_status = visitor.GetResult();
+    }
+    return return_status;
+}
+
+absl::Status CheckAliasKeyword(absl::string_view sql) {
+    return ASTNodeRule([](const zetasql::ASTNode* node,
+        absl::string_view sql) -> absl::Status {
+        if (node->node_kind() == zetasql::AST_ALIAS) {
+            int position = node->GetParseLocationRange()
+                .start().GetByteOffset();
+            if ( sql[position] != 'A' || sql[position+1] != 'S' ) {
+                return absl::Status(
+                    absl::StatusCode::kFailedPrecondition,
+                    absl::StrCat(
+                    "Always use AS keyword for referencing aliases, ",
```

Comment:
I think you can remove the trailing comma here. One can write:

```
std::string foo = "a string "
                  "that didn't fit "
                  "on one line.";
```

Reply:
Also Fixed.

Reply:
ahm, I still see the comma before "In position: "